packages/frontend/src/components/Explorer/CustomMetricModal/FilterForm.tsx

```diff
@@ -0,0 +1,137 @@
+    FieldTarget,
+    FilterRule,
+    isAdditionalMetric,
+    isDimension,
+    isField,
+    isFilterableField,
+    TableCalculation,
+} from '@lightdash/common';
+import { Button, Stack } from '@mantine/core';
+import { Dispatch, FC, SetStateAction, useCallback } from 'react';
+import { useExplorerContext } from '../../../providers/ExplorerProvider';
+import FilterRuleForm from '../../common/Filters/FilterRuleForm';
+import { useFiltersContext } from '../../common/Filters/FiltersProvider';
+import { addFieldRefToFilterRule } from './utils';
+
+export interface MetricFilterRuleWithFieldId
+    extends FilterRule<
+        ConditionalOperator,
+        FieldTarget & { fieldRef: string }
+    > {}
+
+export const FilterForm: FC<{
+    item: Dimension | AdditionalMetric;
+    customMetricFiltersWithIds: MetricFilterRuleWithFieldId[];
+    setCustomMetricFiltersWithIds: Dispatch<
+        SetStateAction<MetricFilterRuleWithFieldId[]>
+    >;
+}> = ({ item, customMetricFiltersWithIds, setCustomMetricFiltersWithIds }) => {
+    const isEditMode = useExplorerContext(
+        (context) => context.state.isEditMode,
+    );
+    const { fieldsMap } = useFiltersContext();
+
+    const dimensions = Object.values(fieldsMap).filter(isDimension);
+
+    const addFieldRule = useCallback(
+        (field: Field | TableCalculation | Dimension | AdditionalMetric) => {
+            if (
+                isAdditionalMetric(field) &&
+                'baseDimensionName' in field &&
+                field.baseDimensionName
+            ) {
+                const baseFieldName = `${field.table}_${field.baseDimensionName}`;
+
+                const baseField = fieldsMap[baseFieldName];
+                const newFilterRule = createFilterRuleFromField(
+                    fieldsMap[baseFieldName],
+                );
+
+                if (isField(baseField) && isFilterableField(baseField)) {
+                    setCustomMetricFiltersWithIds([
+                        ...customMetricFiltersWithIds,
+                        {
+                            ...newFilterRule,
+                            target: {
+                                ...newFilterRule.target,
+                                fieldRef: convertFieldIdToFieldRef(baseField),
+                            },
+                        },
+                    ]);
+                }
+            } else {
+                if (isField(field) && isFilterableField(field)) {
+                    const newFilterRule = createFilterRuleFromField(field);
+
+                    setCustomMetricFiltersWithIds([
+                        ...customMetricFiltersWithIds,
+                        {
+                            ...newFilterRule,
+                            target: {
+                                ...newFilterRule.target,
+                                fieldRef: convertFieldIdToFieldRef(field),
+                            },
+                        },
+                    ]);
+                }
```

Comment:
Since we only support filters targeting dimensions I would remove all the other conditions/logic and keep this block with `if(isDimension(field)) {...}`

Reply:
great point, thanks! i've simplified it here: https://github.com/lightdash/lightdash/pull/5865/commits/280660114ac27d93fbf0ca16c9a5e8b825ebee63

Reply:
can hide the `filters`? 🤷